mrmustard/lab_dev/wires.py

```diff
@@ -15,422 +15,255 @@
+        for m in final[1]:
+            if m in sets[1]:
+                perm.append(lists[1].index(m) + sum(offsets[:1]))
+            else:
+                perm.append(lists[5].index(m) + sum(offsets[:5]))
+        for m in final[2]:
+            if m in sets[2]:
+                perm.append(lists[2].index(m) + sum(offsets[:2]))
+            else:
+                perm.append(lists[6].index(m) + sum(offsets[:6]))
+        for m in final[3]:
+            if m in sets[3]:
+                perm.append(lists[3].index(m) + sum(offsets[:3]))
+            else:
+                perm.append(lists[7].index(m) + sum(offsets[:7]))
+        return w, tuple(perm)
 
     def __repr__(self) -> str:
-        ob_modes, ib_modes, ok_modes, ik_modes = self._args()
-        return f"Wires({ob_modes}, {ib_modes}, {ok_modes}, {ik_modes})"
-
-    def _repr_html_(self):  # pragma: no cover
-        "A matrix plot of the id_array."
-        row_labels = map(str, self._modes)
-        col_labels = ["bra-out", "bra-in", "ket-out", "ket-in"]
-        array = np.abs(self.id_array) / (self.id_array + 1e-15)
-        idxs = (i for i in self.indices)
-        box_size = "60px"  # Set the size of the squares
-        html = '<table style="border-collapse: collapse; border: 1px solid black;">'
-        # colors
-        active = "#5b9bd5"
-        inactive = "#d6e8f7"
-
-        # Add column headers
-        html += "<tr>"
-        for label in [""] + col_labels:  # Add an empty string for the top-left cell
-            html += f'<th style="border: 1px solid black; padding: 5px;">{label}</th>'
-        html += "</tr>"
-
-        # Initialize rows with row labels
-        rows_html = [
-            f'<tr><td style="border: 1px solid black; padding: 5px;">{label}</td>'
-            for label in row_labels
-        ]
-
-        # Add table cells (column by column)
-        for label, col in zip(col_labels, array.T):
-            for row_idx, value in enumerate(col):
-                color = (
-                    "white"
-                    if np.isclose(value, 0)
-                    else (active if np.isclose(value, 1) else inactive)
-                )
-                cell_html = f'<td style="border: 1px solid black; padding: 5px;\
-                      width: {box_size}px; height: {box_size}px; background-color:\
-                          {color}; box-sizing: border-box;'
-                if color == active:
-                    cell_html += (
-                        f' text-align: center; vertical-align: middle;">{str(next(idxs))}</td>'
-                    )
-                else:
-                    cell_html += '"></td>'
-                rows_html[row_idx] += cell_html
-
-        # Close the rows and add them to the HTML table
-        for row_html in rows_html:
-            row_html += "</tr>"
-            html += row_html
-
-        html += "</table>"
-        display(HTML(html))
+        return f"Wires{self.args}"
```

Comment:
https://github.com/XanaduAI/MrMustard/pull/361#discussion_r1518396850

Reply:
not sure I follow 😅